### PR TITLE
Netatmo oh3 update

### PR DESCRIPTION
**Pull request by wzbfyb**

### CODE REVIEW

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java [hunk 1]
```diff
@@ -35,6 +35,8 @@
     private List<NAHomeEvent> events = List.of();
     private List<NAThermProgram> thermSchedules = List.of();
     private List<NAWelcome> cameras = List.of();
+
+    private @Nullable NAObjectMap<NARoom> rooms;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java:39]: I would prefer having this non null with an empty list like the other above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java [hunk 2]
```diff
@@ -44,6 +46,10 @@
         return thermSchedules;
     }
 
+    public void setRooms(NAObjectMap<NARoom> rooms) {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java:49]: Is this of any use somewhere ?

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java [hunk 3]
```diff
@@ -75,6 +81,12 @@ public int getThermSetpointDefaultDuration() {
         return cameras;
     }
 
+    public @Nullable NAObjectMap<NARoom> getRooms() {
+        return rooms;
+    }
+    public @Nullable NARoom getRoom(String key) {
+        return rooms.get(key);
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java:88]: Subject to NPE if rooms is null

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHomeStatus.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ */
+package org.openhab.binding.netatmo.internal.api.dto;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ *
+ * @author Gaël L'hopital - Initial contribution
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHomeStatus.java:21]: Change author to yourself. You did it, be proud of it.

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.api.dto;
+
+import com.google.gson.annotations.SerializedName;
+import org.openhab.binding.netatmo.internal.api.ModuleType;
+
+/**
+ *
+ * @author Gaël L'hopital - Initial contribution
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:20]: Same than above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ */
+
+public class NARoom extends NAModule {
+    private boolean anticipating;
+    private int heating_power_request;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:26]: should be heatingPowerRequest

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    private boolean open_window;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:27]: should be openWindow

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+    private Double therm_measured_temperature;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:28]: same as above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    private String therm_setpoint_mode;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:29]: same as above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    private Double therm_setpoint_temperature;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:30]: same as above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+     * @param anticipating the anticipating to set
+     */
+    public void setAnticipating(boolean anticipating) {
+        this.anticipating = anticipating;
+    }
+
+    /**
+     * @return the heating_power_request
+     */
+    public int getHeating_power_request() {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:52]: this method should be getHeatingPowerRequest()

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        return heating_power_request;
+    }
+
+    /**
+     * @param heating_power_request the heating_power_request to set
+     */
+    public void setHeating_power_request(int heating_power_request) {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:59]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+        this.heating_power_request = heating_power_request;
+    }
+
+    /**
+     * @return the open_window
+     */
+    public boolean isOpen_window() {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:66]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+        return open_window;
+    }
+
+    /**
+     * @param open_window the open_window to set
+     */
+    public void setOpen_window(boolean open_window) {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:73]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+        this.open_window = open_window;
+    }
+
+
+    /**
+     * @return the therm_measured_temperature
+     */
+    public Double getTherm_measured_temperature() {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:81]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+        return therm_measured_temperature;
+    }
+
+    /**
+     * @param therm_measured_temperature the therm_measured_temperature to set
+     */
+    public void setTherm_measured_temperature(Double therm_measured_temperature) {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:88]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+        this.therm_measured_temperature = therm_measured_temperature;
+    }
+
+    /**
+     * @return the therm_setpoint_mode
+     */
+    public String getTherm_setpoint_mode() {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:95]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+        return therm_setpoint_mode;
+    }
+
+    /**
+     * @param therm_setpoint_mode the therm_setpoint_mode to set
+     */
+    public void setTherm_setpoint_mode(String therm_setpoint_mode) {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:102]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+        this.therm_setpoint_mode = therm_setpoint_mode;
+    }
+
+    /**
+     * @return the therm_setpoint_temperature
+     */
+    public double getTherm_setpoint_temperature() {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:109]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+        return therm_setpoint_temperature;
+    }
+
+    /**
+     * @param therm_setpoint_temperature the therm_setpoint_temperature to set
+     */
+    public void setTherm_setpoint_temperature(Double therm_setpoint_temperature) {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NARoom.java:116]: see above

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Body.java [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Bernhard Kreuz - Initial contribution
+ */
+public class Body implements Serializable {
+
+    private Home home;
+    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Body.java:25]: `private Map<String, Object> additionalProperties = new HashMap<>();`

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Body.java [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+    private final static long serialVersionUID = 4318969274668449496L;
+
+    /**
+     * No args constructor for use in serialization
+     * 
+     */
+    public Body() {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Body.java:32]: Not sure this is really needed

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+    private List<Module> modules = new ArrayList<Module>();
+    private List<Room> rooms = new ArrayList<Room>();
+    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
+    private final static long serialVersionUID = -330826367376419729L;
+
+    /**
+     * No args constructor for use in serialization
+     * 
+     */
+    public Home() {
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java:35]: Not sure this is really needed

File: bundles/org.openhab.binding.netatmo/pom.xml [hunk 24]
```diff
@@ -13,5 +13,8 @@
   <artifactId>org.openhab.binding.netatmo</artifactId>
 
   <name>openHAB Add-ons :: Bundles :: Netatmo Binding</name>
+  <properties>
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/pom.xml:16]: You should not skip stat tests

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.Map;
+/**
+ * @author Bernhard Kreuz - Initial contribution
+ */
+public class Home implements Serializable {
+
+    private String id;
+    private List<Module> modules = new ArrayList<Module>();
+    private List<Room> rooms = new ArrayList<Room>();
+    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java:28]: `private Map<String, Object> additionalProperties = new HashMap<>();`

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+/**
+ * @author Bernhard Kreuz - Initial contribution
+ */
+public class Home implements Serializable {
+
+    private String id;
+    private List<Module> modules = new ArrayList<Module>();
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java:26]: `private List<Module> modules = new ArrayList<>();`

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java [hunk 27]
```diff
@@ -0,0 +1,102 @@
...
+    private List<Room> rooms = new ArrayList<Room>();
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Home.java:27]: `private List<Room> rooms = new ArrayList<>();`

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Homestatus.java [hunk 28]
```diff
@@ -0,0 +1,100 @@
...
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+/**
+ * @author Bernhard Kreuz - Initial contribution
+ */
+public class Homestatus implements Serializable {
+
+    private String status;
+    private Integer timeServer;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Homestatus.java:24]: Use primitive type instead

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Homestatus.java [hunk 29]
```diff
@@ -0,0 +1,100 @@
...
+    private Body body;
+    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Homestatus.java:26]: Use diamond operator

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Module.java [hunk 30]
```diff
@@ -0,0 +1,278 @@
...
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+/**
+ * @author Bernhard Kreuz - Initial contribution
+ */
+public class Module implements Serializable {
+
+    private Boolean boilerCable;
+    private Integer firmwareRevision;
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Module.java:24]: use primitive types when possible

File: bundles/org.openhab.binding.netatmo/wie_builden.txt [hunk 31]
```diff
@@ -0,0 +1,3 @@
+mvn clean install -DskipChecks -pl :org.openhab.binding.netatmo
```
Comment (clinique) [inline bundles/org.openhab.binding.netatmo/wie_builden.txt:1]: This should not stay in the repo

**Review by clinique — CHANGES_REQUESTED** [review]
First draft review. Once you'll have reactivate sat tests, you'll find many things to address. Thanks for your work.

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java (hunk 2 above)
Comment (wzbfyb) [reply bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/NAHome.java:49]: not anymore, removed it

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Homestatus.java (hunk 28 above)
Comment (wzbfyb) [reply bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/dto/energy/Homestatus.java:24]: thanks, most of the classes I am editing now where actually from a code generator :)

**Pull request closed without merging**